### PR TITLE
lp1743256: Synchronize execution of Vamp analyzers

### PR DESCRIPTION
**Pull request by uklotzde**
https://bugs.launchpad.net/mixxx/+bug/1743256

Just a workaround, we can't fix Vamp ourselves.

Cherry-picked from #1624

### CODE REVIEW

File: src/analyzer/vamp/vamppluginadapter.cpp [hunk 1]
```diff
@@ -137,37 +134,133 @@ void initPluginPaths() {
 #endif
 }
 
-void initPluginLoader() {
-    initPluginPaths();
-    s_pPluginLoader = Vamp::HostExt::PluginLoader::getInstance();
+std::mutex s_mutex;
```
Comment (daschuer) [inline src/analyzer/vamp/vamppluginadapter.cpp]: It looks like there are some reasons to prefer QMutex over std::mutex:
https://stackoverflow.com/questions/29170495/performance-of-stdmutex-and-qmutex-in-mingw-64-posix-thread-version

Comment (uklotzde) [reply src/analyzer/vamp/vamppluginadapter.cpp]: http://man7.org/linux/man-pages/man7/pthreads.7.html
"...In NPTL, thread synchronization primitives (mutexes, thread joining, and so on) are implemented using the Linux futex(2) system call."

I have more trust in the standard implementation that comes with a broader background and audience. I suppose QMutex will once become a simple wrapper around std::mutex. There is no more reason to maintain a custom solution. That's legacy and history.

Comment (rryan) [reply src/analyzer/vamp/vamppluginadapter.cpp]: Do QObject::thread, QObject::moveToThread work with std::thread? How do you get a QThread* handle to a std::thread to pass to those?

What about queued connections where the received "lives" in a std::thread?

File: src/analyzer/vamp/vamppluginadapter.cpp [hunk 2]
```diff
@@ -137,37 +134,133 @@ void initPluginPaths() {
...
+
+Vamp::HostExt::PluginLoader* s_pluginLoader = nullptr;
+
+Vamp::HostExt::PluginLoader* getPluginLoader(
+        const std::lock_guard<std::mutex>&) {
```
[the remark this answers is not in the document]
Comment (uklotzde) [reply src/analyzer/vamp/vamppluginadapter.cpp]: To enforce locking the mutex before invoking this function. The parameter ensures that the function can only be called from within a locked scope. Suggestions other than using a simple (and unchecked) comment? We could add a comment to the parameter if that helps.

[the remark this answers is not in the document]
Comment (daschuer) [reply src/analyzer/vamp/vamppluginadapter.cpp]: This unused variable is more confusing than it helps. Please revert it to a comment. 
This is anyway in an anonymous namespace, so we do not need to worry too much about misuse. 

.. or use the thread_annotations.h. Unfortunately I was not able to wrap my hand around it.
@rryan can you explain what we need to do here?

**Comment by uklotzde** [comment]
Using the annotations would require to move the mutex as a static member into the header. That's not worth it.

**Comment by daschuer** [comment]
Can't we just remove the unused mutex valiable? It is IMHO more confusing than it helps.

**Comment by daschuer** [comment]
Thank you! LGTM

**Pull request merged**